tools/sync_bugzilla_to_github.py

```diff
@@ -0,0 +1,285 @@
+# description of status in bugzilla will be pushed to github, but any changes
+# in github will not be reflected back in bugzilla.
+
+import re
+import os
+import urllib.parse
+
+import requests
+from github import Github
+
+GH_REPO = 'mozilla/application-services'
+GH_LABEL = 'bugzilla'
+
+BZ_URL = 'https://bugzilla.mozilla.org/rest'
+
+SYNCED_ISSUE_TEXT = '\n\n---\n\N{LADY BEETLE} Issue is synchronized with Bugzilla [Bug {id}](https://bugzilla.mozilla.org/show_bug.cgi?id={id})\n'
+SYNCED_ISSUE_BUGID_REGEX = re.compile('\N{LADY BEETLE} Issue is synchronized with Bugzilla \\[Bug (\\d+)\\]')
+
+# Jira adds some metadata to issue descriptions, indicated by this separator.
+# We want to preserve any lines like this from the github issue description.
+JIRA_ISSUE_MARKER = '\N{BOX DRAWINGS LIGHT TRIPLE DASH VERTICAL}'
+
+# For now, only look at recent bugs in order to preserve sanity.
+LAST_CHANGE_TIME = '20190801'
+
+
+def log(msg, *args, **kwds):
+    msg = str(msg)
+    print(msg.format(*args, **kwds))
+
+
+def get_json(url):
+    """Fetch a URL and return the result parsed as JSON."""
+    r = requests.get(url)
+    r.raise_for_status()
+    return r.json()
+
+
+class BugSet(object):
+  """A set of bugzilla bugs, which we might like to mirror into GitHub.
+  
+  This class knows how to query the bugzilla API to find bugs, and how to
+  fetch appropriate metadata for mirroring into github.
+  
+  Importantly, it knows how to use a bugzilla API key to find confidential
+  or security-sensitive bugs, and can repor their existence without leaking
+  potentially confidential details (e.g. by reporting a placeholder summary
+  of "confidential issue" rather than the actual bug summary).
+
+  Use `update_from_bugzilla()` to query for bugs and add them to an in-memory
+  store, then access them as if this were a dict keyed by bug number.  Each bug
+  will be a dict with the following fields:
+
+      * `id`: The bug id, as an integer
+      * `whiteboard`: The bug's whiteboard text, as a string
+      * `is_open`: Whether the bug is open, as a boolean
+      * `summary`: The one-line bug summry, as a string
+      * `status`: The bug's status field, as a string
+      * `comment0`: The bug's first comment, which is typically a longer description, as a string
+  """
+
+  def __init__(self, api_key=None):
+    self.api_key = api_key
+    self.bugs = {}
+
+  def __getitem__(self, key):
+    return self.bugs[str(key)]
+  
+  def __iter__(self):
+    return iter(self.bugs)
+
+  def __len__(self):
+    return len(self.bugs)
+
+  def update_from_bugzilla(self, **kwds):
+    """Slurp in bugs from bugzilla that match the given query keywords."""
+    # First, fetch a minimal set of "safe" metadata that we're happy to put in
+    # a public github issue, even for confidential bugzilla bugs.
+    # This is the only query that's allowed to use a BZ API token to access
+    # confidential bug info.
+    url = BZ_URL + '/bug?include_fields=id,whiteboard,is_open'
```

Comment:
not clear to me why "whiteboard" is in this list?

Reply:
Ah, nice catch; for FxA we will slurp in any bug with `[fxa]` in its whiteboard tag and ignore bugs with `[fxa-waffle-ignore]` but I haven't implemented that here yet, so it can probably be removed for now.